python/ray/train/data_parallel_trainer.py

```diff
@@ -220,6 +220,11 @@ def __init__(self, train_loop_per_worker, my_backend_config:
         TuneCheckpointManager
     ] = _DataParallelCheckpointManager
 
+    # Exposed here for testing purposes. Should never need
+    # to be overriden.
+    _backend_executor_cls: Type[BackendExecutor] = BackendExecutor
+    _training_iterator_cls: Type[TrainingIterator] = TrainingIterator
```

Comment:
if this is just for testing purporses, we can use patch.object instead in the tests.

```
from unittest.mock import patch
with patch.object(ray.train._internal.backend, "BackendExecutor", new_backend_executor_cls):
          ...
```

Reply:
I am not sure patching will work here as it needs to also touch the Trainable in a separate process.